Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISLoadAndProcess.py

```diff
@@ -236,6 +237,19 @@ def _getRunFromADSOrNone(self, run, isTrans):
         # Not found
         return None
 
+    def _collapse_workspace_groups(self, workspaces, ungrouped_workspaces):
+        """Given a list of workspaces, which themselves could be groups of workspaces,
+        return a new list of workspaces which are TOF"""
+        for ws_name in workspaces:
+            ws = AnalysisDataService.retrieve(ws_name)
+            if isinstance(ws, WorkspaceGroup):
+                self._collapse_workspace_groups(ws.getNames(), ungrouped_workspaces)
+                AnalysisDataService.remove(ws_name)
```

Comment:
I think we might have a problem if all of the child workspaces are in wavelength, then they will not be regrouped into the `TOF` group but their original group will be removed anyway, meaning they will be ejected into the top level of the ADS. I think we should keep them in the original group. We could add a check here to check if any of the child workspaces in the hierarchy are non-TOF and if so don't remove the group (that check could perhaps also be done by `_collapse_workspace_groups` since we are walking down the hierarchy anyway).

I think this will only happen if all of the inputs are in wavelength, otherwise we will throw a "mixed units" error earlier on and won't get this far. We should add a test for this first to see if it gets this far, though, because I'm not sure if it will succeed anyway.